### PR TITLE
Update CI, remove some warnings

### PR DESCRIPTION
**Pull request by greglandrum**
Some of this is a response to CI configuration changes.
Adjusting to that ended up resulting in a newer version of clang for the OSX builds and that then required some code cleanup to get rid of some warnings.

I don't think there's anything really substantive here.

### CODE REVIEW

**Comment by bp-kelley** [comment]
@greglandrum  it looks like the Mac build got set to "warnings are errors"

File: Code/GraphMol/new_canon.cpp [hunk 1]
```diff
@@ -170,7 +170,6 @@ void compareRingAtomsConcerningNumNeighbors(Canon::canon_atom *atoms,
     memset(lastLevelNbrs.get(), 0, nAtoms * sizeof(char));
     memset(currentLevelNbrs.get(), 0, nAtoms * sizeof(char));
     memset(revisitedNeighbors.get(), 0, nAtoms * sizeof(int));
-    unsigned count = 1;
```
Comment (bp-kelley) [inline Code/GraphMol/new_canon.cpp:173]: Did you mean for this to be included here?

**Review by bp-kelley — CHANGES_REQUESTED** [review]

File: Code/GraphMol/new_canon.cpp (hunk 1 above)
Comment (bp-kelley) [reply Code/GraphMol/new_canon.cpp:173]: [-Werror,-Wunused-but-set-variable]


Is this the new g++11 default?

Comment (greglandrum) [reply Code/GraphMol/new_canon.cpp:173]: The errors are from clang-16. I've addressed them later in the PR

**Comment by greglandrum** [comment]
> @greglandrum it looks like the Mac build got set to "warnings are errors"

yes, we added that for the linux and mac builds in #6675

File: Code/GraphMol/new_canon.cpp (hunk 1 above)
Comment (bp-kelley) [reply Code/GraphMol/new_canon.cpp:173]: Of course it's clang (OSX and all)

**Pull request merged**